src/app/core/sync-status/background-processing-indicator/background-processing-indicator.component.ts

```diff
@@ -1,81 +1,85 @@
-import {
-  Component,
-  Input,
-  OnChanges,
-  SimpleChanges,
-  ViewChild,
-} from "@angular/core";
+import { Component, Input, OnInit, ViewChild } from "@angular/core";
 import { MatMenuTrigger } from "@angular/material/menu";
 import { BackgroundProcessState } from "../background-process-state.interface";
+import { Observable } from "rxjs";
+import { map, startWith } from "rxjs/operators";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
 
 /**
  * A dumb component handling presentation of the sync indicator icon
  * and an additional details dropdown listing all currently running background processes.
  */
+@UntilDestroy()
 @Component({
   selector: "app-background-processing-indicator",
   templateUrl: "./background-processing-indicator.component.html",
   styleUrls: ["./background-processing-indicator.component.scss"],
 })
-export class BackgroundProcessingIndicatorComponent implements OnChanges {
+export class BackgroundProcessingIndicatorComponent implements OnInit {
   /** details on current background processes to be displayed to user */
-  @Input() backgroundProcesses: BackgroundProcessState[] = [];
+  @Input() backgroundProcesses: Observable<BackgroundProcessState[]>;
+  filteredProcesses: Observable<BackgroundProcessState[]>;
+  taskCounterObservable: Observable<number>;
+  allTasksFinished: Observable<boolean>;
 
   /** whether processes of with the same title shall be summarized into one line */
   @Input() summarize: boolean = true;
 
-  /** how many special tasks (e.g. index creations) are currently being processed */
-  taskCounter: number;
-
   /** handle to programmatically open/close the details dropdown */
   @ViewChild(MatMenuTrigger) taskListDropdownTrigger: MatMenuTrigger;
 
-  constructor() {}
-
-  ngOnChanges(changes: SimpleChanges) {
-    if (!changes.backgroundProcesses) {
-      return;
-    }
-
-    if (this.summarize && this.backgroundProcesses) {
-      this.backgroundProcesses = this.summarizeProcesses(
-        this.backgroundProcesses
-      );
-    }
-
-    this.taskCounter = this.backgroundProcesses?.filter(
-      (s) => s.pending
-    ).length;
+  ngOnInit() {
+    this.filteredProcesses = this.backgroundProcesses.pipe(
+      map((processes) => this.summarizeProcesses(processes))
+    );
+    this.taskCounterObservable = this.filteredProcesses.pipe(
+      map((processes) => processes.filter((p) => p.pending).length)
+    );
+    this.allTasksFinished = this.taskCounterObservable.pipe(
+      startWith(0),
+      map((tc) => tc === 0)
+    );
+    this.taskCounterObservable
+      .pipe(untilDestroyed(this))
+      .subscribe((amount) => {
+        if (amount === 0) {
+          this.taskListDropdownTrigger.closeMenu();
+        } else {
+          this.taskListDropdownTrigger.openMenu();
```

Comment:
Personally I dont really like this behavior because it means that every time something is saved the popup opens. Most oft the users are probably quite irritated by that. I think just showing the sync icon with the potential to open it is sufficient